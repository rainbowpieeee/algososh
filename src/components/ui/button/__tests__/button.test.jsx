import { render, screen, fireEvent } from "@testing-library/react";

import renderer from "react-test-renderer";

import { Button } from "../button";

const buttonText = "Run Jest";

const alertText = "rendered ok";

describe("checking button with snapshots.", () => {
  it("button should be rendered with text", () => {
    const tree = renderer.create(<Button text={buttonText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("button should be rendered without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("disabled button should be rendered", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("button should be rendered with loader", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("checking button event listener", () => {
  it("click on button works and callback is ok", () => {
    window.alert = jest.fn();

    render(
      <Button
        text={buttonText}
        onClick={() => {
          alert(alertText);
        }}
      />
    );

    const button = screen.getByText(buttonText);

    fireEvent.click(button);

    expect(window.alert).toHaveBeenCalledWith(alertText);
  });

  it("click on button works", () => {
    const fn = jest.fn();
    render(<Button text={buttonText} onClick={fn} />);

    const button = screen.getByText(buttonText);

    fireEvent.click(button);
    fireEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(2);
  });
});