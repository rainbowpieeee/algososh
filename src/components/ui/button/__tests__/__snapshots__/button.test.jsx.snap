// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking button with snapshots. button should be rendered with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`checking button with snapshots. button should be rendered with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Run Jest
  </p>
</button>
`;

exports[`checking button with snapshots. button should be rendered without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`checking button with snapshots. disabled button should be rendered 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
